pkg/prepare/prepare.go

```diff
@@ -168,13 +177,49 @@ func (p *Prepare) ConfigureCoreDNS(clusterDomain, maeshNamespace string) error {
 		return err
 	}
 
-	if err := p.restartPods(deployment); err != nil {
+	if err := p.RestartPods(deployment); err != nil {
 		return err
 	}
 
 	return nil
 }
 
+// backupConfigMap backs up a configmap with `-backup` appended to its name, and returns the updated configmap.
+func (p *Prepare) backupConfigMap(configMap *corev1.ConfigMap, maeshNamespace string) (*corev1.ConfigMap, error) {
+	// Create a copy of the configmap, but with new objectmeta to avoid conflicts.
+	newConfigMap := &corev1.ConfigMap{
+		TypeMeta: configMap.TypeMeta,
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      configMap.Name + "-backup",
+			Namespace: maeshNamespace,
+		},
+		Data:       configMap.Data,
+		BinaryData: configMap.BinaryData,
+	}
+
+	if _, err := p.client.KubernetesClient().CoreV1().ConfigMaps(newConfigMap.Namespace).Get(newConfigMap.Name, metav1.GetOptions{}); err == nil {
+		// Backup already exists, delete it.
+		if err = p.client.KubernetesClient().CoreV1().ConfigMaps(newConfigMap.Namespace).Delete(newConfigMap.Name, &metav1.DeleteOptions{}); err != nil {
+			return nil, err
+		}
+	}
+
+	// Create backup configmap.
+	if _, err := p.client.KubernetesClient().CoreV1().ConfigMaps(newConfigMap.Namespace).Create(newConfigMap); err != nil {
+		return nil, err
+	}
+
+	// Label the configmap as backed up.
+	if configMap.ObjectMeta.Labels == nil {
+		configMap.ObjectMeta.Labels = make(map[string]string)
+	}
+
+	configMap.ObjectMeta.Labels["maesh-backed-up"] = "true"
```

Comment:
I feel a little anxious every time I see this kind of magic string used everywhere in the code. 
Why not create a constant?

Reply:
Same comment the `maesh-patched` string.